sap-abap/services/cloudwatch/zcl_aws1_cwt_actions.clas.testclasses.abap

```diff
@@ -0,0 +1,612 @@
+    DATA lo_alarm_list_result TYPE REF TO /aws1/cl_cwtdescralarmsoutput.
+    DATA lv_bucket_name TYPE /aws1/s3_bucketname.
+    DATA lv_uuid_16 TYPE sysuuid_x16.
+
+    CONSTANTS cv_metric_name  TYPE  /aws1/cwtmetricname VALUE 'NumberOfObjects'.
+    CONSTANTS cv_namespace  TYPE  /aws1/cwtnamespace VALUE 'AWS/S3'.
+    CONSTANTS cv_comparison_operator  TYPE  /aws1/cwtcomparisonoperator VALUE 'GreaterThanThreshold'.
+    CONSTANTS cv_statistic  TYPE  /aws1/cwtstatistic VALUE 'Average'.
+    CONSTANTS cv_threshold  TYPE  /aws1/rt_double_as_string VALUE 10.
+    CONSTANTS cv_alarm_description  TYPE  /aws1/cwtalarmdescription VALUE 'Alarm when number of objects exceeds 10'.
+    CONSTANTS cv_actions_enabled  TYPE  /aws1/cwtactionsenabled VALUE 'X'.
+    CONSTANTS cv_evaluation_periods TYPE  /aws1/cwtevaluationperiods VALUE 1.
+    CONSTANTS cv_unit TYPE /aws1/cwtstandardunit VALUE 'Percent'.
+    CONSTANTS cv_period TYPE /aws1/cwtperiod VALUE 86400.
+    CONSTANTS cv_bucket_name TYPE /aws1/s3_bucketname VALUE 'code-example-cwt-'.
+
+    "Create an S3 bucket
+    lv_uuid_16 = cl_system_uuid=>create_uuid_x16_static( ).
+    lv_bucket_name = cv_bucket_name && lv_uuid_16.
+    TRANSLATE lv_bucket_name TO LOWER CASE.
+    ao_s3->createbucket( iv_bucket = lv_bucket_name ).
+
+    "Define alarm name
+    lv_alarm_name = 'code-example-cwt-s3-alarm-' && lv_uuid_16.
+    TRANSLATE lv_alarm_name TO LOWER CASE.
+
+    "Create S3 dimensions
+    CREATE OBJECT lo_dimensions
+      EXPORTING
+        iv_name  = 'StorageType'
+        iv_value = 'AllStorageTypes'.
+    INSERT lo_dimensions INTO TABLE lt_dimensions.
+
+    CREATE OBJECT lo_dimensions
+      EXPORTING
+        iv_name  = 'BucketName'
+        iv_value = lv_bucket_name.
+    INSERT lo_dimensions INTO TABLE lt_dimensions.
+
+    ao_cwt->putmetricalarm(
+      EXPORTING
+        iv_alarmname          = lv_alarm_name
+        iv_metricname         = cv_metric_name
+        iv_namespace           = cv_namespace
+        iv_comparisonoperator = cv_comparison_operator
+        iv_statistic           = cv_statistic
+        iv_threshold           = cv_threshold
+        iv_alarmdescription   = cv_alarm_description
+        iv_actionsenabled     = cv_actions_enabled
+        iv_evaluationperiods  = cv_evaluation_periods
+        iv_unit                = cv_unit
+        iv_period              = cv_period
+        it_dimensions          = lt_dimensions
+        ).
+
+    "Testing disable_alarm_actions
+    CREATE OBJECT lo_alarmname EXPORTING iv_value = lv_alarm_name.
+    INSERT lo_alarmname INTO TABLE lt_alarmnames.
+
+    ao_cwt_actions->disable_alarm_actions( it_alarm_names = lt_alarmnames ).
+
+    "Validation
+    lo_alarm_list_result = ao_cwt->describealarms(
+      EXPORTING it_alarmnames = lt_alarmnames
+    ).
+
+    lv_found = abap_false.
+
+    LOOP AT lo_alarm_list_result->get_metricalarms( ) INTO DATA(lo_alarms).
+      IF lo_alarms->get_actionsenabled( ) = ' '.
```

Comment:
This compares to 'X' above? Oh because it's assert ing it is is disabled. (Just needs an ack)